calico_containers/adapter/datastore.py

```diff
@@ -211,6 +219,45 @@ def from_json(cls, json_str):
         return rules
 
 
+class BGPPeer(object):
+    """
+    Class encapsulating a BGPPeer.
+    """
+
+    def __init__(self, ip, as_num):
+        """
+        Constructor.
+        :param ip: The BGPPeer IP address (string or IPAddress)
+        :param as_num: The AS Number (string or int)
+        """
+        self.ip = IPAddress(ip)
+        self.as_num = int(as_num)
+
+    def to_json(self):
+        """
+        Convert the BGPPeer to a JSON string.
+        :return: A JSON string.
+        """
+        json_dict = {"ip": str(self.ip), "as_num": self.as_num}
+        return json.dumps(json_dict)
+
+    @classmethod
+    def from_json(cls, json_str):
+        """
+        Convert the json string into a BGPPeer object.
+        :param json_str: The JSON string representing a BGPPeer.
+        :return: A BGPPeer object.
+        """
+        json_dict = json.loads(json_str)
+        return cls(json_dict["ip"], json_dict["as_num"])
+
+    def __eq__(self, other):
+        if not isinstance(other, BGPPeer):
+            return NotImplemented
```

Comment:
I've pushed a fix.  Apologies for missing that!  I've re-run code coverage and I can't see any lines from this PR that are not being hit.